prow/cmd/clonerefs/main.go

```diff
@@ -154,12 +156,28 @@ func main() {
 					logrus.Fatalf("Clone specification for %s/%s found both in Prow variables and user-provided flags", jobRefs.Refs.Org, jobRefs.Refs.Repo)
 				}
 			}
-			results = append(results, clone.Run(jobRefs.Refs, o.srcRoot, o.gitUserName, o.gitUserEmail, o.aliases.aliases))
+			wg.Add(1)
+			go func(out chan<- clone.Record, group *sync.WaitGroup) {
+				defer group.Done()
+				out <- clone.Run(jobRefs.Refs, o.srcRoot, o.gitUserName, o.gitUserEmail, o.aliases.aliases)
+			}(output, wg)
 		}
 	}
 
+	wg.Add(len(o.refs.gitRefs))
 	for _, gitRef := range o.refs.gitRefs {
-		results = append(results, clone.Run(gitRef, o.srcRoot, o.gitUserName, o.gitUserEmail, o.aliases.aliases))
+		go func(out chan<- clone.Record, group *sync.WaitGroup) {
+			defer group.Done()
+			out <- clone.Run(gitRef, o.srcRoot, o.gitUserName, o.gitUserEmail, o.aliases.aliases)
+		}(output, wg)
```

Comment:
You don't need to pass `output` or `wg` to the goroutine here or the goroutine that handles refs found in the `$JOB_SPEC` above. They will be captured appropriately. 

I think you *do* need to pass `gitRef` though because its value changes each loop iteration so there is a race with the created goroutines reading it.

Reply:
Yes, you need to pass gitRef

Reply:
Ahh good catch @cjw

Reply:
Originally was passing the channel so that it could be unidirectional inside of the func but yeah that doesn't really matter